core/consensus/authority/impl/authority_manager_impl.cpp

```diff
@@ -84,12 +84,24 @@ namespace kagome::authority {
                                     bool finalized) {
     auto node = getAppropriateAncestor(block);
 
+    std::shared_ptr<ScheduleNode> adjusted_node;
+
     if (not node) {
-      return AuthorityManagerError::ORPHAN_BLOCK_OR_ALREADY_FINALISED;
+      if (not directChainExists(block, root_->block)) {
+        return AuthorityManagerError::ORPHAN_BLOCK_OR_ALREADY_FINALISED;
+      }
+      // a fix that helps to run node from accidentally created db state
+      // when finalizing block is earlier than scheduled node block
+      // may be removed when sure that node doesn't fail
+      // because of justification errors
+      adjusted_node = ScheduleNode::createAsRoot(block);
+      adjusted_node->actual_authorities =
+          std::make_shared<primitives::AuthorityList>(
+              *root_->actual_authorities);
```

Comment:
I think it is so durty hack and we shoul not use it. Better to log problem and terminate node